lib/log.js

```diff
@@ -0,0 +1,94 @@
+var path = require('path')
+var fs = require('fs')
+var Console = require('console')
+var mkdir = require('mkdirp')
+var wsock = require('websocket-stream')
+var http = require('http')
+var through = require('through2')
+
+var nodeLogger = null
+var browserLogger = null
+
+var PORT = 17509
+
+module.exports.Node = function () {
+  if (!nodeLogger) {
+    nodeLogger = setupNodeLogger()
+  }
+  return nodeLogger
+}
+
+module.exports.Browser = function () {
+  if (!browserLogger) {
+    browserLogger = setupBrowserLogger()
+  }
+  return browserLogger
+}
+
+function setupNodeLogger () {
+  var electron = require('electron')
+
+  // create log dir
+  var logDir = path.join(electron.app.getPath('userData'), 'data', 'logs')
+  mkdir.sync(logDir)
+
+  // get electron app data dir
+  var filename = (new Date()).toISOString() + '.txt'
+  var logFilename = path.join(logDir, filename)
+
+  // fs write stream to file
+  var dest = fs.createWriteStream(logFilename)
+
+  // through stream to multiplex to stdout
+  var multi = multiplex([process.stdout, dest])
+
+  // create console instance
+  var log = new Console.Console(multi, null).log
+
+  // start ws server
+  websocketServer(function (socket) {
+    socket.on('data', function (text) {
+      multi.write(text)
+    })
+  })
+
+  return log
+}
+
+function setupBrowserLogger () {
+  var socket = wsock('ws://localhost:'+PORT)
```

Comment:
What would happen if the socket server is not running when the browser tries to log?